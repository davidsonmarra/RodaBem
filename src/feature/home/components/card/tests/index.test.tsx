import React from 'react';
import {act, create} from 'react-test-renderer';
import Card from '..';
import {TouchableOpacity} from 'react-native';

const mockData = {
  name: 'Name',
  brand: 'Brand',
  kmDriven: '100',
  kmPerLiter: '10',
};

const createComponent = ({data = mockData, onPress = jest.fn(), ...rest}) =>
  create(<Card onPress={onPress} data={data} {...rest} />);

describe('Card', () => {
  it('renders correctly', () => {
    const component = createComponent({});
    expect(component).toMatchSnapshot();
  });

  it('should renders name', () => {
    const component = createComponent({});
    const title = component.root.findByProps({children: 'Name'});
    expect(title).toBeTruthy();
  });

  it('should renders brand', () => {
    const component = createComponent({});
    const description = component.root.findByProps({children: 'Brand'});
    expect(description).toBeTruthy();
  });

  it('should renders kmDriven', () => {
    const component = createComponent({});
    const kmDriven = component.root.findByProps({children: '100 Km rodados'});
    expect(kmDriven).toBeTruthy();
  });

  it('should renders kmPerLiter', () => {
    const component = createComponent({});
    const kmPerLiter = component.root.findByProps({children: '100 Km/L'});
    expect(kmPerLiter).toBeTruthy();
  });

  it('should call onPress when press card', () => {
    const onPress = jest.fn();
    const component = createComponent({onPress}).root;
    const buton = component.findByType(TouchableOpacity);

    act(() => {
      buton.props.onPress();
    });

    expect(onPress).toHaveBeenNthCalledWith(1, mockData);
  });
});
