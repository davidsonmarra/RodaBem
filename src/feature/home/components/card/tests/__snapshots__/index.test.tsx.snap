// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E5E5E5",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#121212",
            "fontSize": 24,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Name
    </Text>
    <Text
      style={
        [
          {
            "color": "#222222",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Brand
    </Text>
    <View
      style={
        {
          "height": 12,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#222222",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      100 Km rodados
    </Text>
  </View>
  <View
    style={
      {
        "marginLeft": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#121212",
            "fontSize": 24,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      100 Km/L
    </Text>
  </View>
</View>
`;
